flang/lib/Lower/CharacterRuntime.cpp

```diff
@@ -88,3 +88,20 @@ Fortran::lower::genCharCompare(Fortran::lower::AbstractConverter &converter,
   return genRawCharCompare(converter, loc, cmp, lhsBuffer, fir::getLen(lhs),
                            rhsBuffer, fir::getLen(rhs));
 }
+
+void Fortran::lower::genTrim(Fortran::lower::FirOpBuilder &builder,
+                             mlir::Location loc, const mlir::Value &resultBox,
+                             const mlir::Value &stringBox) {
```

Comment:
We shouldn't pass mlir:;Value types this way. They are meant to be passed like `int`.

Reply:
Thanks, serves me right for copy pasting from an interface using ExtendedValues and not changing that.